src/ducks/Signals/SignalMaster.js

```diff
@@ -36,61 +50,137 @@ export class SignalMaster extends React.PureComponent {
+          return triggerSettingsFormData.isActive
+            ? 'Create public signal'
+            : 'Create private signal'
+        }
+        default: {
+          return ''
+        }
+      }
+    }
+
     return (
       <div className={styles.wrapper}>
-        {step === STEPS.SETTINGS && (
-          <TriggerForm
-            settings={formProps}
-            metaFormSettings={this.props.metaFormSettings}
-            onSettingsChange={this.handleSettingsChange}
-          />
-        )}
-        {step === STEPS.CONFIRM && (
-          <AboutForm
-            {...meta}
-            isEdit={this.props.isEdit}
-            onBack={this.backToSettings}
-            onSubmit={this.handleAboutFormSubmit}
-          />
-        )}
+        <Icon
+          className={styles.closeButton}
+          onClick={this.props.onClose}
+          type='close'
+        />
+        <Panel header={getTitle()} className={styles.TriggerPanel}>
+          {step === STEPS.SETTINGS && (
+            <TriggersForm
+              triggers={[currentTrigger]}
+              settings={triggerSettingsFormData}
+              canRedirect={this.props.canRedirect}
+              metaFormSettings={metaFormSettings}
+              onSettingsChange={this.handleSettingsChange}
+            />
+          )}
+          {step === STEPS.CONFIRM && (
+            <AboutForm
+              triggerMeta={triggerAboutFormData}
+              isEdit={this.props.isEdit}
+              onBack={this.backToSettings}
+              onSubmit={this.handleAboutFormSubmit}
+            />
+          )}
+
+          <div className={styles.triggerToggleBlock}>
+            <Toggle
+              onClick={toggleSignalCallback}
+              isActive={currentTrigger.isActive}
+            />
+            <div className={styles.triggerToggleLabel}>
+              {currentTrigger.isActive ? 'Public' : 'Private'}
+            </div>
+          </div>
+        </Panel>
       </div>
     )
   }
 
-  backToSettings = () => {
-    this.setState({ step: STEPS.SETTINGS })
+  backToSettings = prefilledData => {
+    const { trigger } = this.state
+    this.setState({
+      step: STEPS.SETTINGS,
+      trigger: { ...trigger, ...prefilledData }
+    })
   }
 
   handleSettingsChange = formProps => {
+    const { trigger } = this.state
     this.setState({
       trigger: mapFormPropsToTrigger(
         formProps,
-        (this.props.trigger || {}).trigger
+        (this.props.trigger || {}).trigger || trigger
       ),
       step: STEPS.CONFIRM
     })
   }
 
   handleAboutFormSubmit = about => {
+    const data = {
+      ...this.state.trigger,
+      ...about,
+      ...{ shouldReload: this.props.canRedirect }
+    }
+
     if (this.props.isEdit) {
-      this.props.updateTrigger({ ...this.state.trigger, ...about })
+      this.props.updateTrigger(data)
     } else {
-      this.props.createTrigger({ ...this.state.trigger, ...about })
+      this.props.createTrigger(data)
     }
```

Comment:
`this.props.createTrigger(data)` will be called in both cases

Reply:
Why?